BitwardenShared/UI/Platform/Application/Views/BitwardenBadge.swift

```diff
@@ -12,18 +12,34 @@ struct BitwardenBadge: View {
     /// Scaling is relative to the title style so it doesn't add quite as much padding with
     /// the larger font sizes.
     @ScaledMetric(relativeTo: .title)
-    var badgePadding: CGFloat = 8
+    var badgePadding: CGFloat = 2
 
     /// The value to display in the badge.
     let badgeValue: String
 
+    /// The size of the rendered text in the badge.
+    @SwiftUI.State var textSize = CGSize.zero
+
+    // MARK: Computed Properties
+
+    /// The diameter of the circle badge background based on the text size.
+    var circleDiameter: CGFloat {
+        max(textSize.width, textSize.height)
+    }
+
     // MARK: View
 
     var body: some View {
         Text(badgeValue)
-            .styleGuide(.callout, weight: .bold, includeLineSpacing: false)
+            .styleGuide(.callout, weight: .bold, includeLinePadding: false, includeLineSpacing: false)
             .foregroundStyle(Asset.Colors.iconBadgeForeground.swiftUIColor)
-            .padding(badgePadding)
+            .onSizeChanged { textSize = $0 }
+            // Add custom horizontal and vertical padding to ensure the frame of the badge is a
+            // square. When the view is clipped to a circle, this ensures the frame of the view
+            // matches the size of the circle. Otherwise, clipping can cause there to be extra
+            // space in the frame which can affect the layout when the badge is used.
+            .padding(.horizontal, (circleDiameter - textSize.width) / 2 + badgePadding)
+            .padding(.vertical, (circleDiameter - textSize.height) / 2 + badgePadding)
```

Comment:
Without these changes, the frame of the badge had extra vertical padding when displaying text that was taller than wider. This was also causing alignment issues at the top of action card.

| Before | After |
| --- | --- |
| <img width="113" alt="Screenshot 2024-09-27 at 4 41 10 PM" src="https://github.com/user-attachments/assets/220adaf2-6c84-4b65-8f2a-24e8707e6140"> | <img width="88" alt="Screenshot 2024-09-27 at 4 43 19 PM" src="https://github.com/user-attachments/assets/02369ff5-f2a9-4662-bae1-9d9e8ee69a09"> |